### ISSUE TITLE
pre-rednered text doesn't changed if input changes

### ISSUE THREAD

**Issue by Tunous** [divadretlaw/EmojiText#5]
Due to the way how SwiftUI View state works there is an issue in EmojiText where, when I modify text provided via initializer it will stay unchanged.

The problem appears when identity of EmojiText doesn't change but the user provides input that might change. The way how pre-rendering works is that it will store modified raw input in internal state. Unfortunately that internal state will survive even if users provide different values keeping previously displayed text unchanged.

A workaround is to explicitly change identity of EmojiText from call site, but it would be better if this library handled input changes automatically.

Below is example code that reproduces the issue. Tapping on "Switch" button doesn't change visible text unless I uncomment the line with id.

```swift
struct Preview: View {
    @State private var text = "mastodon :mastodon:"

    private let emojis: [any CustomEmoji] = [
        RemoteEmoji(shortcode: "mastodon", url: URL(string: "https://files.mastodon.social/custom_emojis/images/000/003/675/original/089aaae26a2abcc1.png")!),
        SFSymbolEmoji(shortcode: "iphone")
    ]

    var body: some View {
        VStack {
            EmojiText(verbatim: text, emojis: emojis)
                //.id(text) // Uncommenting this line fixes issue

            Button("Switch") {
                if text.starts(with: "m") {
                    text = "iphone :iphone:"
                } else {
                    text = "mastodon :mastodon:"
                }
            }
        }
    }
}
```

I tried experimented with a way to resolve it internally by adding `onChange(of: raw)` inside of EmojiText but that doesn't seem to work for initial change. I'm not sure why and don't have any other ideas currently so I'm submitting this as issue instead of pull request.

Similar issue might happen if provided emojis change but I didn't check that case.

**Comment by divadretlaw** [comment]
Should be implemented in `2.2.0`. Please let me know if it works or there are any issues

**Comment by divadretlaw** [comment]
Should be fixed in `2.5.1`
